FrontEnd/src/components/Table.tsx

```diff
@@ -0,0 +1,59 @@
+import { render } from '@testing-library/react';
+import React from 'react';
+import data from '../data.json';
+import '../stories/Table.css';
+
+export interface TableProps {
+  error: boolean;
+  errorMessage: { [index: number]: string };
+}
+
+const headers = [
+  { label: 'Tag' },
+  { label: 'Subquadrat' },
+  { label: 'SpCode' },
+  { label: 'DBH' },
+  { label: 'Htmeas' },
+  { label: 'Codes' },
+  { label: 'Comments' },
+];
+
+export default function Table({ error, errorMessage }: TableProps) {
+  return (
+    <div>
+      {
+        <>
+          <table>
+            <thead>
+              <tr>
+                {headers.map((row) => {
+                  // display headers in table row using html array map
+                  return <td>{row.label}</td>;
+                })}
+              </tr>
+            </thead>
+          </table>
+
+          <div>
+            {data.map((data, index) => {
+              return (
+                <table>
+                  <tr>
+                    <td>{data.Tag}</td>
+                    <td>{data.Subquadrat}</td>
+                    <td>{data.SpCode}</td>
+                    <td>{data.DBH}</td>
+                    <td>{data.Htmeas}</td>
+                    <td>{data.Codes}</td>
+                    <td>{data.Comments}</td>
+                  </tr>
+                  <tr className="errorMessage">{errorMessage[index]}</tr>
```

Comment:
If `errorMessage[index]` is not defined, how does it look visually?

Reply:
Here is what the table looks like without any errors:
![2022-07-22 (3)](https://user-images.githubusercontent.com/90858408/180501544-cbcd270c-304e-4438-8ee9-8c97b91f0310.png)

Here is what the table looks like with errors on rows 0, 3, and 5
![2022-07-22 (1)](https://user-images.githubusercontent.com/90858408/180501545-248a7aca-2327-4ec1-a033-c1da291c7053.png)

Reply:
Looks great!